lib/lobstersbot/plugins/title.rb

```diff
@@ -0,0 +1,78 @@
+require 'net/http'
+
+require 'lobstersbot/plugins/title/wikipedia'
+require 'lobstersbot/plugins/title/lobsters'
+
+module Lobstersbot
+  module Title
+    @@handlers = [
+      WikipediaHandler.new,
+      LobstersHandler.new,
+    ]
+
+    StopIteration = Class.new(StandardError)
+
+    def parse_title(response)
+      nread = 0
+      title = nil
+
+      response.read_body do |chunk|
+        raise StopIteration if nread > 16_384
+
+        title = chunk[/<title>([^<]+)<\/title>/im, 1]&.strip
+        raise StopIteration if title
+
+        nread += chunk.length
+      end
+    rescue StopIteration
+      title
+    end
+    module_function :parse_title # rubocop:disable Style/AccessModifierDeclarations
+
+    def fetch(uri, limit = 10, &block)
```

Comment:
I think I'd move this logic out to be independent of the `Title` command, also means you don't need to make the above a module function maybe?

Reply:
Done.